bundles/org.openhab.binding.ipobserver/src/main/java/org/openhab/binding/ipobserver/internal/IpObserverDiscoveryService.java

```diff
@@ -0,0 +1,142 @@
+            Arrays.asList(THING_WEATHER_STATION));
+    private ExecutorService discoverySearchPool = scheduler;
+    private HttpClient httpClient;
+
+    @Activate
+    public IpObserverDiscoveryService(@Reference HttpClientFactory httpClientFactory) {
+        super(SUPPORTED_THING_TYPES_UIDS, 240);
+        httpClient = httpClientFactory.getCommonHttpClient();
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    protected HttpClient getHttpClient() {
+        return httpClient;
+    }
+
+    public void submitDiscoveryResults(String ip) {
+        ThingUID thingUID = new ThingUID(THING_WEATHER_STATION, ip.replace('.', '_'));
+        HashMap<String, Object> properties = new HashMap<>();
+        properties.put("address", ip);
+        thingDiscovered(DiscoveryResultBuilder.create(thingUID).withProperties(properties).withLabel("Weather Station")
+                .withRepresentationProperty("address").build());
+    }
+
+    private void scanSingleSubnet(InterfaceAddress hostAddress) {
+        byte[] broadcastAddress = hostAddress.getBroadcast().getAddress();
+        // Create subnet mask from length
+        int shft = 0xffffffff << (32 - hostAddress.getNetworkPrefixLength());
+        byte oct1 = (byte) (((byte) ((shft & 0xff000000) >> 24)) & 0xff);
+        byte oct2 = (byte) (((byte) ((shft & 0x00ff0000) >> 16)) & 0xff);
+        byte oct3 = (byte) (((byte) ((shft & 0x0000ff00) >> 8)) & 0xff);
+        byte oct4 = (byte) (((byte) (shft & 0x000000ff)) & 0xff);
+        byte[] subnetMask = new byte[] { oct1, oct2, oct3, oct4 };
+        // calc first IP to start scanning from on this subnet
+        byte[] startAddress = new byte[4];
+        startAddress[0] = (byte) (broadcastAddress[0] & subnetMask[0]);
+        startAddress[1] = (byte) (broadcastAddress[1] & subnetMask[1]);
+        startAddress[2] = (byte) (broadcastAddress[2] & subnetMask[2]);
+        startAddress[3] = (byte) (broadcastAddress[3] & subnetMask[3]);
+        // Loop from start of subnet to the broadcast address.
+        for (int i = ByteBuffer.wrap(startAddress).getInt(); i < ByteBuffer.wrap(broadcastAddress).getInt(); i++) {
+            try {
+                InetAddress currentIP = InetAddress.getByAddress(ByteBuffer.allocate(4).putInt(i).array());
+                // Try to reach each IP with a timeout of 500ms which is enough for local network
+                if (currentIP.isReachable(500)) {
+                    String host = currentIP.getHostAddress().toString();
+                    logger.debug("Unknown device was found at: {}", host);
+                    discoverySearchPool.execute(new IpObserverDiscoveryJob(this, host));
+                }
+            } catch (IOException e) {
+            }
+        }
+    }
+
+    @Override
+    protected void startScan() {
+        discoverySearchPool = Executors.newFixedThreadPool(DISCOVERY_THREAD_POOL_SIZE);
```

Comment:
This thread pool needs to be shutdown, when the scan is stopped.

Reply:
DONE.